Source/Scene/Model/GeoJsonLoader.js

```diff
@@ -240,146 +272,182 @@ const geoJsonObjectTypes = {
-      defaultValue.EMPTY_OBJECT
-    );
-    for (const propertyId in featureProperties) {
-      if (featureProperties.hasOwnProperty(propertyId)) {
-        if (!defined(properties[propertyId])) {
-          properties[propertyId] = new Array(featureCount);
-        }
-      }
-    }
-  }
+  const featureCount = features.length;
 
-  // Fill in the property values. Default to empty string for undefined values.
   for (let i = 0; i < featureCount; i++) {
-    const feature = result.features[i];
-    for (const propertyId in properties) {
-      if (properties.hasOwnProperty(propertyId)) {
-        const value = defaultValue(feature.properties[propertyId], "");
-        properties[propertyId][i] = value;
+    const feature = features[i];
+    if (defined(feature.lines)) {
+      const linesLength = feature.lines.length;
+      for (let j = 0; j < linesLength; j++) {
+        const line = feature.lines[j];
+        vertexCount += line.length;
+        indexCount += (line.length - 1) * 2;
       }
     }
   }
 
-  const jsonMetadataTable = new JsonMetadataTable({
-    count: featureCount,
-    properties: properties,
-  });
-
-  const propertyTable = new PropertyTable({
-    id: 0,
-    count: featureCount,
-    jsonMetadataTable: jsonMetadataTable,
-  });
-  const propertyTables = [propertyTable];
-
-  const schema = new MetadataSchema({});
-
-  const structuralMetadata = new StructuralMetadata({
-    schema: schema,
-    propertyTables: propertyTables,
-  });
+  // Allocate typed arrays
+  const positionsTypedArray = new Float32Array(vertexCount * 3);
+  const featureIdsTypedArray = new Float32Array(vertexCount);
+  const indicesTypedArray = IndexDatatype.createTypedArray(
+    vertexCount,
+    indexCount
+  );
+  const indexDatatype = IndexDatatype.fromTypedArray(indicesTypedArray);
 
-  // Find the cartographic bounding box
-  const cartographicMin = new Cartesian3(
+  // Process the data. Convert positions to local ENU. Generate indices.
+  const localMin = new Cartesian3(
     Number.POSITIVE_INFINITY,
     Number.POSITIVE_INFINITY,
     Number.POSITIVE_INFINITY
   );
 
-  const cartographicMax = new Cartesian3(
+  const localMax = new Cartesian3(
     Number.NEGATIVE_INFINITY,
     Number.NEGATIVE_INFINITY,
     Number.NEGATIVE_INFINITY
   );
 
+  let vertexCounter = 0;
+  let segmentCounter = 0;
+
   for (let i = 0; i < featureCount; i++) {
-    const feature = result.features[i];
-    const linesLength = feature.lines.length;
-    for (let j = 0; j < linesLength; j++) {
-      const line = feature.lines[j];
-      const positionsLength = line.length;
-      for (let k = 0; k < positionsLength; k++) {
-        Cartesian3.minimumByComponent(
-          cartographicMin,
-          line[k],
-          cartographicMin
-        );
-        Cartesian3.maximumByComponent(
-          cartographicMax,
-          line[k],
-          cartographicMax
-        );
+    const feature = features[i];
+    if (defined(feature.lines)) {
```

Comment:
Since this loop is rather nested, let's negate the if statement to flatten it out one level:

```js
if (!defined(feature.lines)) {
  continue;
}
```